R/target_market_share.R

```diff
@@ -127,102 +127,62 @@ target_market_share <- function(data,
 
   if (weight_production) {
     data <- summarize_weighted_production(
       data,
-      !!!rlang::syms(summary_groups),
-      use_credit_limit = use_credit_limit
+      !!!rlang::syms(summary_groups)
     )
+
   } else {
     data <- summarize_unweighted_production(
       data,
-      .data$sector_ald,
-      .data$technology,
-      .data$year,
       !!!rlang::syms(summary_groups)
     )
   }
 
-  reweighting_groups <- maybe_add_name_ald(
-    c("sector_ald", "region", "scenario", "scenario_source", "year"),
-    by_company
-  )
-
-  data <- reweight_technology_share(
-    data,
-    !!!rlang::syms(reweighting_groups)
-  )
-
   if (nrow(data) == 0) {
     return(empty_target_market_share_output())
   }
 
-  target_groups <- c("sector_ald", "scenario", "year", "region")
+  target_groups <- c("sector_ald", "scenario", "region", "name_ald")
 
-  initial_sector_summaries <- data %>%
-    maybe_group_by_name_ald(target_groups, by_company = by_company) %>%
-    summarize(sector_weighted_production = sum(.data$weighted_production)) %>%
+  data <- data %>%
+    group_by(!!!rlang::syms(c(target_groups, "year"))) %>%
+    mutate(sector_weighted_production = sum(.data$weighted_production)) %>%
     arrange(.data$year) %>%
-    maybe_group_by_name_ald(
-      c("sector_ald", "scenario", "region"),
-      by_company = by_company
-    ) %>%
-    filter(row_number() == 1L) %>%
-    rename(
-      initial_sector_production = .data$sector_weighted_production
-    ) %>%
-    select(-.data$year)
+    group_by(!!!rlang::syms(target_groups)) %>%
+    mutate(initial_sector_production = first(.data$sector_weighted_production))
 
-  initial_technology_summaries <- data %>%
-    maybe_group_by_name_ald(
-      c(target_groups, "technology"),
-      by_company = by_company
-    ) %>%
-    summarize(
-      technology_weighted_production = sum(.data$weighted_production)
-    ) %>%
+  data <- data %>%
+    group_by(!!!rlang::syms(c(target_groups, "technology", "year"))) %>%
+    mutate(technology_weighted_production = sum(.data$weighted_production)) %>%
     arrange(.data$year) %>%
-    maybe_group_by_name_ald(
-      c("sector_ald", "technology", "scenario", "region"),
-      by_company = by_company
-    ) %>%
-    filter(row_number() == 1L) %>%
-    rename(
-      initial_technology_production = .data$technology_weighted_production
-    ) %>%
-    select(-.data$year)
+    group_by(!!!rlang::syms(c(target_groups, "technology"))) %>%
+    mutate(initial_technology_production = first(.data$technology_weighted_production))
 
   data <- data %>%
-    left_join(
-      initial_sector_summaries,
-      by = maybe_add_name_ald(
-        c("sector_ald", "scenario", "region"),
-        by_company = by_company
-      )
-    ) %>%
-    left_join(
-      initial_technology_summaries,
-      by = maybe_add_name_ald(
-        c("sector_ald", "scenario", "region", "technology"),
-        by_company = by_company
-      )
-    ) %>%
```

Comment:
Before: this chunk was done in a really awkward way. I calculate two data-frames from `data`: `inital_sector_summaries` and `initial_technology_summaries`, and then `join` them back into the original `data`

Now: This is done more elegantly with a couple calls to `group_by` and `summarize`